css-viewport/Overview.bs

```diff
@@ -356,6 +360,157 @@ only the value previously set to it.
+		followed by a percentage character ("%") which indicates a zoom factor multiplied by 100.
+		A 0 percentage is treated as if it was 100%.
+
+		Note: The treatment of 0 is a web compatibility quirk.
+
+</dl>
+
+Negative values for 'zoom' are illegal.
+
+<div class="example">
+	Example of the 'zoom' property applied during hover for magnification effect.
+
+	<pre>
+		&lt;div class="messageBox"&gt;
+			&lt;div class="label"&gt;Text of the label&lt;/div&gt;
+		&lt;/div&gt;
+
+		&lt;style&gt;
+		.messageBox {
+		    width: 10em;
+		    padding: 2em;
+		    border: medium solid lightblue;
+		}
+
+		.messageBox:hover {
+		    zoom: 150%;
+		}
+
+		.label {
+		    background: lightgrey;
+		    padding: 1em;
+		    text-align: center;
+		}
+		&lt;/style&gt;
+	</pre>
+
+	Here is an llustration of the before and after hover state of the message box element:
+	<img src="css_zoom_hover_example.png" alt="Two images,
+	showing the zooming effect before and after zoom has applied. The second is 1.5 larger.">
+</div>
+
+<div class="example">
+	Example of nested zoom.
+	In this example, "Inner text" is 4x as large as "Outer text",
+	and "Middle text" is 2x as large as "Outer text".
+
+	<pre>
+		&lt;div style="zoom: 2"&gt;
+		  Middle text
+			&lt;div style="zoom: 2"&gt;
+			   Inner text
+			&lt;div&gt;
+		&lt;div&gt;
+		Outer text
+	</pre>	 
+</div>
+
+<div class="example">
+	Example of replaced elements. In this example,
+	the image and iframe will be twice as large as their default sizing.
+
+	<pre>
+	&lt;div style="zoom: 2"&gt;
+		&lt;img src="..."&gt;
+		&lt;iframe src="..."&gt;&lt;/iframe&gt;
+	&lt;div&gt;
+	</pre>	 
+</div>
+
+The <dfn>effective zoom</dfn> of an element is the product of its computed
+value of 'zoom' and all ancestors' computed values of 'zoom'.
+
+The <dfn export>scaled</dfn> value of a CSS length is the used value of that length;
+in particular it includes zoom.
+
+The <dfn export>unscaled</dfn> value of a CSS length relative to an element is the [=scaled=] value divided by the element's [=effective zoom=].
+
+<div class="note">
+	The [=effective zoom=] of an element in a nested frame may be a value other than 1 even if 'zoom' is never set on an element in that frame. This can be observed by authors via APIs such as {{Window/devicePixelRatio}} and {{Element/getBoundingClientRect}}.
+</div>
+
+<h3 id='zoom-om'>
+	DOM and CSSOM interaction
+</h3>
+
+	Computed style APIs (i.e., all values returned by {{getComputedStyle()}}) that are non-auto and non-percentage lengths must be [=unscaled=].
+
+	The {{Element/getBoundingClientRect}},
+	{{Element/getClientRects}},
+	and {{IntersectionObserver}} APIs must return rects with [=scaled=]
+	lengths.
+
+	All other APIs related to element geometries must return [=unscaled=] lengths.
+	This is explained in detail in [[cssom-view#extensions-to-the-htmlelement-interface]].
+
+	The {{Window/devicePixelRatio}} of a frame is multiplied by the [=effective zoom=] inherited by its parent frame.
```

Comment:
So that means the effective zoom of the iframe element affects the CSS reference pixel of the iframe document. I guess that means getBoundingClientRect() etc on elements inside the iframe will return px sizes for that new reference pixel?

Reply:
Makes sense.